services/venice-server/src/main/java/com/linkedin/venice/listener/grpc/handlers/GrpcOutboundResponseHandler.java

```diff
@@ -0,0 +1,59 @@
+package com.linkedin.venice.listener.grpc.handlers;
+
+import static io.netty.handler.codec.http.HttpResponseStatus.*;
+
+import com.google.protobuf.ByteString;
+import com.linkedin.davinci.listener.response.ReadResponse;
+import com.linkedin.venice.compression.CompressionStrategy;
+import com.linkedin.venice.grpc.GrpcErrorCodes;
+import com.linkedin.venice.listener.ServerStatsContext;
+import com.linkedin.venice.listener.grpc.GrpcHandlerContext;
+import com.linkedin.venice.protocols.VeniceServerResponse;
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import io.netty.handler.codec.http.HttpResponseStatus;
+
+
+public class GrpcOutboundResponseHandler extends VeniceServerGrpcHandler {
+  @Override
+  public void processNext(GrpcHandlerContext ctx) {
+    ByteBuf body;
+    CompressionStrategy compressionStrategy = CompressionStrategy.NO_OP;
+
+    ReadResponse obj = ctx.getReadResponse();
+    ServerStatsContext statsContext = ctx.getGrpcStatsContext();
+    VeniceServerResponse.Builder veniceServerResponseBuilder = ctx.getVeniceServerResponseBuilder();
+    if (!ctx.hasError()) {
+      compressionStrategy = obj.getCompressionStrategy();
+
+      veniceServerResponseBuilder.setCompressionStrategy(compressionStrategy.getValue());
+      veniceServerResponseBuilder.setResponseRCU(obj.getRCU());
+      veniceServerResponseBuilder.setIsStreamingResponse(obj.isStreamingResponse());
+
+      if (obj.isFound()) {
+        body = obj.getResponseBody();
+        veniceServerResponseBuilder.setSchemaId(obj.getResponseSchemaIdHeader());
+        statsContext.setResponseStatus(OK);
+      } else {
+        ctx.setError();
+        body = Unpooled.EMPTY_BUFFER;
+        statsContext.setResponseStatus(NOT_FOUND);
+        veniceServerResponseBuilder.setErrorCode(GrpcErrorCodes.KEY_NOT_FOUND);
+        veniceServerResponseBuilder.setErrorMessage("Key not found");
+      }
+    } else {
+      statsContext.setResponseStatus(HttpResponseStatus.BAD_REQUEST);
+      body = Unpooled.EMPTY_BUFFER;
+    }
+
+    byte[] array = new byte[body.readableBytes()];
+    body.getBytes(body.readerIndex(), array);
+    veniceServerResponseBuilder.setData(ByteString.copyFrom(array))
+        .setCompressionStrategy(compressionStrategy.getValue());
+
+    if (ctx.hasError())
```

Comment:
let's use braces. For an empty response can we create a static byte array and use it everytime we want to send error response. Something like
```java 
public static byte[] EMPTY_BUFFER = new byte[0]
```

Reply:
Is there a benefit to this? `Unpooled.EMPTY_BUFFER` would achieve the same functionality as using the above method. .